### DISCUSSION TITLE
Changes to data option on node

### DISCUSSION THREAD

**Discussion by mdowton** [bcakmakoglu/vue-flow#1245] [Q&A] [answered]
I have a data property on `node` , but when I update that property the `onNodesChange` event doesn't fire ?
```
data: structuredClone(defaultNodeData[newNodeType]) ?? {
        input: {},
        private: {},
        public: {},
      },
```

**Comment by bcakmakoglu** [comment, marked as answer]
`onNodesChange` doesn't listen to data obj changes.
It's used to inform you of changes like adding, removing, position or select.

[You can find the type info here.](https://vueflow.dev/typedocs/types/NodeChange.html)

**Reply by bcakmakoglu** [reply]
You would have to use your own method of informing about these changes - VueFlow doesn't really care much about the custom data you use.

**Reply by mdowton** [reply]
ok thanks @bcakmakoglu
